cli/dcoscli/data/config-schema/core.json

```diff
@@ -9,7 +9,7 @@
     },
     "dcos_url": {
       "type": "string",
-      "pattern": "^(?:(?:https?)://)(?:(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\\.?)+(?:[a-zA-Z]{2,6}\\.?|[a-zA-Z0-9-]{2,}\\.?)?|\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})(?::\\d+)?(?:/?|[/?]\\S+)$",
+      "pattern": "^(?:(?:https?)://)(([a-zA-Z0-9_-])+(:[a-zA-Z0-9_-]+)?@){0,1}(?:(?:[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?\\.?)+(?:[a-zA-Z]{2,6}\\.?|[a-zA-Z0-9-]{2,}\\.?)?|\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3})(?::\\d+)?(?:/?|[/?]\\S+)$",
```

Comment:
I'm pretty sure username/password in the URL is plaintext, and `requests` does the base64 encoding.  have you tested this against a real basic auth server?